src/Footer/Footer.tsx

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames';
+import { FilterType } from '../types/FilterType';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  filterType: string;
+  setfilterType: (type: string) => void;
+  visibleTodos: Todo[];
+  todos: Todo[];
+  handleDelete: (id: number)=> void;
+};
+
+export const Footer: React.FC<Props> = ({
+  filterType,
+  setfilterType,
+  visibleTodos,
+  todos,
+  handleDelete,
+}) => {
+  const clearCompleted = () => todos
+    .forEach(todo => {
+      if (todo.completed) {
+        handleDelete(todo.id);
+      }
+    });
+
+  const isActive = visibleTodos.filter(todo => !todo.completed);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${isActive.length} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames('filter__link', {
+            'filter__link selected': filterType === FilterType.All,
+          })}
+          onClick={() => setfilterType(FilterType.All)}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          className={classNames('filter__link', {
+            'filter__link selected': filterType === FilterType.Active,
+          })}
+          onClick={() => setfilterType(FilterType.Active)}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          className={classNames('filter__link', {
+            'filter__link selected': filterType === FilterType.Completed,
+          })}
+          onClick={() => setfilterType(FilterType.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      <button
+        data-cy="ClearCompletedButton"
+        type="button"
+        className="todoapp__clear-completed"
+        onClick={clearCompleted}
+      >
+        {(todos.some(todo => todo.completed))
+         && 'Clear completed'}
```

Comment:
Check how this button works on [the working example](https://mate-academy.github.io/react_todo-app-with-api/) and yours.
<img width="904" alt="Screenshot 2022-10-10 at 19 59 08" src="https://user-images.githubusercontent.com/17098736/194918273-358997cc-2859-4833-a3a8-6eb779931958.png">